CONTRIBUTING.md

```diff
@@ -108,6 +109,35 @@ The following environment variables are used for [CSFLE testing](https://github.
  * `KMS_TLS_CA_FILE`
  * `KMS_TLS_CERTIFICATE_KEY_FILE`
 
+### Updating spec tests
+
+Tests from the MongoDB Specifications repository are included through a
+submodule and updated automatically through Dependabot. To update tests
+manually, switch to the `tests/specifications` directory and update the
+repository to the appropriate commit. Remember to commit this change to the
+library repository.
+
+#### Handling test failures on updates
+
+Failures on updates can occur for multiple reasons, and the remedy to this will
+depend on the type of failure. Note that only tests for implemented
+specifications are run in the test runner.
+
+ * If a specification is not fully implemented (e.g. a recent change to the spec
+   has not been applied yet), skip the test in question with a reference to the
+   ticket that covers the change
+ * If a test fails because it uses features not yet implemented in the unified
+   test runner, skip the corresponding test with a reference to the ticket that
+   covers implementing the new features
+ * If the test failure points to a bug in the spec, consider the effort required
+   to fix the failure. If it's a small change, commit and push the fix directly
+   to the pull request. Otherwise, skip the test with a reference to a ticket to
+   fix the failing test.
+
+The goal is that the library passes tests with the latest spec version at all
+times, either by implementing small changes quickly, or by skipping tests as
+necessary.
```

Comment:
@jmikola this is a first shot at covering potential failures in a dependabot PR. Let me know if I missed something or if we should change the guidance. Either way, this would be a learning experience from us and might require some care when implementing spec changes (e.g. don't modify existing test cases when adding functionality).

Reply:
The three bullets above seem sufficient.

I'm not sure what you mean by "don't modify existing test cases when adding functionality". I assume you're not talking about making custom modifications to the JSON files, as I've never done that (although I recall libmongoc has).

Reply:
I'm referring to when changes are made in the specs repository. For example, new functionality should rely on new tests as much as possible, without changing existing tests. This allows skipping new tests without sacrificing code coverage because we'd have to skip a test that we were previously able to run.